addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -688,28 +688,10 @@ options.Class.include({
         }
         for (const widget of widgets) {
             const methodsNames = widget.getMethodsNames();
-            const specialMethodsNames = [];
-            // If it's a pageOption, it's most likely to need to reload, so check the widgets.
-            if (this.data.pageOptions) {
-                specialMethodsNames.push(methodsNames);
-            } else {
-                for (const methodName of methodsNames) {
-                    if (this.specialCheckAndReloadMethodsNames.includes(methodName)) {
-                        specialMethodsNames.push(methodName);
-                    }
-                }
-            }
-            if (!specialMethodsNames.length) {
-                continue;
-            }
-            let paramsReload = false;
-            for (const methodName of specialMethodsNames) {
-                if (widget.getMethodsParams(methodName).reload) {
-                    paramsReload = true;
-                    break;
-                }
-            }
-            if (paramsReload) {
+            const methodNamesToCheck = this.data.pageOptions
+                ? methodsNames
+                : methodsNames.filter(m => this.specialCheckAndReloadMethodsNames.includes(m));
+            if (methodNamesToCheck.some(m => widget.getMethodsParams(m).reload)) {
                 return true;
             }
```

Comment:
I was wondering, in the case pageOptions is true, we could just do getMethodsParams once without a methodName, since `data-reload` is common to all methods. This *might* be better for stable, as before, the methodName wouldn't really have returned anything extra, but might now? It's a bit of a stretch though.

I was thinking something like
```js
            if (this.data.pageOptions && widget.getMethodsParams().reload) {
                return true;
            }
            if (
                widget.getMethodsNames()
                    .filter(methodName => this.specialCheckAndReloadMethodsNames.includes(methodName))
                    .some(methodName => widget.getMethodsParams(methodName).reload)
            ) {
                return true;
            }
 ```
 
 Let me know what you think 🤔

Reply:
Good thinking but I answered that already in the commit message :wink: 

> [...] and allow overrides to make reload a specific parameter for a specific method without the need of data-reload added in the DOM.

(as explained, I could have just target master really but I chose it was better to consolidate the API in stable given the incredibly low risk of breaking anything and given the fact that if I do break something... it's just a reload of the editor)